examples/protobuf/vision_receiver_example.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file multicast_receiver_example.cpp
+ *
+ * @author Lucas Haug <lucas.haug@thuneratz.org>
+ * @author Lucas Schneider <lucas.schneider@thuneratz.org>
+ *
+ * @brief Example on how to use the MulticastReceiver
+ *
+ * @date 04/2021
+ *
+ * @copyright MIT License - Copyright (c) 2021 ThundeRatz
+ */
+
+#include <iostream>
+#include <string>
+#include <google/protobuf/util/json_util.h>
+#include <google/protobuf/text_format.h>
+#include <boost/asio.hpp>
+#include "boost/bind.hpp"
+
+#include "travesim_adapters/protobuf/vision_sender.hpp"
+#include "travesim_adapters/udp/multicast_receiver.hpp"
+
+/*****************************************
+ * Private Constants
+ *****************************************/
+
+#define BUFFER_SIZE 1024U
+
+/*****************************************
+ * Main Function
+ *****************************************/
+
+int main(int argc, char* argv[]) {
+    const std::string listen_address_str = "0.0.0.0";
+    const std::string multicast_address_str = "224.0.0.1";
+    const short multicast_port = 10002;
+
+    char data_buff[BUFFER_SIZE];
+
+    try {
+        travesim::udp::MulticastReceiver my_receiver(multicast_address_str, multicast_port);
+
+        size_t data_size = 0;
+
+        for (long int i = 0;; i++) {
+            data_size = my_receiver.receive(data_buff, BUFFER_SIZE);
+
+            if (data_size > 0) {
+                fira_message::sim_to_ref::Environment env_data;
+                env_data.ParseFromString(data_buff);
+
+                std::string parsed_msg;
+
+                google::protobuf::util::JsonPrintOptions options;
+                options.add_whitespace = true;
+                options.always_print_primitive_fields = true;
+
+                google::protobuf::util::MessageToJsonString(env_data, &parsed_msg, options);
+
+                std::cout << parsed_msg << std::endl;
+            }
+
+            if (i % 100000 == 0) {
+                std::cout << "Loop count: " << i << std::endl;
+            }
+        }
+    } catch (std::exception& e) {
+        std::cerr << "Exception: " << e.what() << "\n";
+    }
+
+    return 0;
+}
```

Comment:
Eu não entendi de onde veio esse exemplo haushaushaushaus, porque a gente não recebe nada de protobuf da visão aqui nos adapters

Reply:
Esse exemplo serve pra conferir se o vision adapter está funcionando certinho. Eu só peguei o código do multicast example e troquei as configs pra se conectar ao vision adapter (inclusive esqueci de mudar o nome do arquivo na docstring >.<)

Reply:
Ahhhh tendi, eh pra isso eu tava usando os arquivos de [teste da visão do próprio fira_thundervolt](https://github.com/ThundeRatz/fira_thundervolt/blob/develop/integration_tests/test_vision.py) (cuidado que lá ficou pra printar o z da bola, mas se o z for 0 vai bugar), porque aí dá pra ter certeza que os códigos dos times tão funcionando com esse código que a gente fez aqui pra UDP/protobuf.

Reply:
O problema é que o fira_thundervolt é um projeto privado, né hahahahahaha